### ISSUE TITLE
OS/X (bash) - Command to remove non essential files not working

### ISSUE THREAD

**Issue by jeetendra-choudhary** [angular/quickstart#369]
I am using mac with Sierra and the command to remove non essential files not working, here is the output for same - 

`Jeetendras-MacBook-Pro:quickstart Jeet$ xargs -a non-essential-files.txt rm -rf`
` xargs: illegal option -- a`
`usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]`
            ` [-L number] [-n number [-x]] [-P maxprocs] [-s size]`
            ` [utility [argument ...]]`

Though there are workarounds for it, it would be good to have a fix here.

**Comment by leebenson** [comment]
I've submitted a [PR](https://github.com/angular/angular.io/pull/3160)

Workaround:

> cat non-essential-files.txt | xargs rm -rf

**Comment by sakovias** [comment]
Not sure what I'm doing wrong but neither
``` 
cat non-essential-files.txt | xargs rm -rf
```
nor
```
xargs -a non-essential-files.txt rm -rf
```
do anything for me. I'm on Ubuntu with bash. Any thoughts?

Also, this output looks a bit weird
```
$ xargs -a non-essential-files.txt
README.mdr.config.jstxt
```

**Comment by leebenson** [comment]
you could also do:

`rm -rf $(cat non-essential-files.txt)`

... which, AFAIK, expands the same way all *nix systems.

Of course, always be sure you know exactly what you're executing when it comes to `rm -rf`... if '/' happens to be in the file for some odd reason, and you're a sudo user, you could wipe out your entire system in one go!

**Comment by trescenzi** [comment]
There are some extra nonreadable characters, at least by RHEL7, within the file. 

`rm -rf $(cat non-essential-files.txt)` doesn't work either.

A `hexdump` of the file shows what's wrong:

```
(0)[11:06][tcrescenzi@unixdeva22]~/web/frameworks-next/angular2-listbox/(master)λ hexdump non-essential-files.txt -C
00000000  2e 67 69 74 0d 0a 2e 67  69 74 69 67 6e 6f 72 65  |.git...gitignore|
00000010  0d 0a 2e 74 72 61 76 69  73 2e 79 6d 6c 0d 0a 2a  |...travis.yml..*|
00000020  2e 73 70 65 63 2a 2e 74  73 0d 0a 43 48 41 4e 47  |.spec*.ts..CHANG|
00000030  45 4c 4f 47 2e 6d 64 0d  0a 65 32 65 0d 0a 66 61  |ELOG.md..e2e..fa|
00000040  76 69 63 6f 6e 2e 69 63  6f 0d 0a 6b 61 72 6d 61  |vicon.ico..karma|
00000050  2e 63 6f 6e 66 2e 6a 73  0d 0a 6b 61 72 6d 61 2d  |.conf.js..karma-|
00000060  74 65 73 74 2d 73 68 69  6d 2e 6a 73 0d 0a 4c 49  |test-shim.js..LI|
00000070  43 45 4e 53 45 0d 0a 6e  6f 6e 2d 65 73 73 65 6e  |CENSE..non-essen|
00000080  74 69 61 6c 2d 66 69 6c  65 73 2e 74 78 74 0d 0a  |tial-files.txt..|
00000090  70 72 6f 74 72 61 63 74  6f 72 2e 63 6f 6e 66 69  |protractor.confi|
000000a0  67 2e 6a 73 0d 0a 52 45  41 44 4d 45 2e 6d 64 0d  |g.js..README.md.|
000000b0  0a                                                |.|
000000b1
```

The solution should be as simple as copying the file to your clipboard and then pasting it over the existing file. That worked for me.

Here's also a list without the offending characters you can past that will work with `xargs -a <file> rm -rf`

```
.git
.gitignore
.travis.yml
*.spec*.ts
CHANGELOG.md
e2e
favicon.ico
karma.conf.js
karma-test-shim.js
LICENSE
non-essential-files.txt
protractor.config.js
README.md
```

For those uncertain if it's working try `xargs -a <file> ls` and it'll show you what will be touched by the command.

**Comment by leebenson** [comment]
good catch @trescenzi 

Although, to be clear, `-a` is not a valid xargs switch on OS X (at least Sierra). So even if the encoding is fixed, it'll still fail.

`rm -rf $(cat non-essential-files.txt)` works as-is on my Mac, even with the current format.

So I'd suggest either:

1. Fixing the file, and making `rm -rf $(cat non-essential-files.txt)` the official *nix blurb
2. Having separate instructions for OS X / other *nix flavours where `xargs -a` is valid

**Comment by trescenzi** [comment]
@leebenson I think `1.` makes the most sense. Sounds like you'd have to fix the file regardless and it's probably better to use a command that's valid on more *nix flavors(including OS X). 

Thanks for the quick response!

**Comment by wardbell** [comment]
Closed by 816b8bd87736de0b2874502dbd38f531a281d472
Works on my MBP. Let me know if it fails for you
